app/controllers/dashboard/protocols_controller.rb

```diff
@@ -140,9 +140,10 @@ def edit
   end
 
   def update
-    @protocol = @protocol.becomes(params[:protocol][:type].constantize) unless @protocol_type.nil?
-    if params[:updated_protocol_type] == 'true' && params[:protocol][:type] == 'Study'
-      @protocol.update_attribute(:type, params[:protocol][:type])
+    protocol_type = params[:protocol][:type]
+    @protocol = @protocol.becomes(protocol_type.constantize) unless protocol_type.nil?
+    if params[:updated_protocol_type] == 'true' && protocol_type == 'Study'
+      @protocol.update_attribute(:type, protocol_type)
       @protocol.activate
       @protocol = Protocol.find(params[:id]) #Protocol reload
```

Comment:
Shouldn't `@protocol.reload` do the trick here? It's class was already changed on line 144. As is, this line is weird.